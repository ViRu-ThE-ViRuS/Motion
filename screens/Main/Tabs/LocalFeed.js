import React, { Component } from 'react';

import {
    StyleSheet,
    View,
    Text,
    BackHandler,
} from 'react-native';

import ActionButton from 'react-native-action-button';
import Icon from 'react-native-vector-icons/MaterialIcons';
import ICON from 'react-native-vector-icons/MaterialCommunityIcons';
import NavBarComponent from './../../../components/NavBarComponent/NavBarComponent';

export default class LocalFeed extends Component {
    static navigationOptions = {
        title: 'Local Feed',
        headerRight: (<NavBarComponent />),
    }

    componentWillMount() {
        BackHandler.addEventListener('hardwareBackPress', BackHandler.exitApp);
    }

    componentWillUnmount() {
        BackHandler.removeEventListener('hardwareBackPress', BackHandler.exitApp);
    }

    createMotionScreen() {
        this.props.navigation.navigate('createMotion');
    }

    render() {
        return (
          <View style={styles.container}>
            <Text>
                Local
            </Text>

            <ActionButton buttonColor="#5C7D87">
              <ActionButton.Item buttonColor="#02c3bd" title="Settings">
                <Icon name="settings" style={styles.actionButtonIcon} />
              </ActionButton.Item>
              <ActionButton.Item buttonColor="#1abc9c" title="Filter">
                <ICON name="filter" style={styles.actionButtonIcon} />
              </ActionButton.Item>
              <ActionButton.Item buttonColor="#1b998b" title="Create Motion" onPress={() => this.createMotionScreen()}>
                <Icon name="add" style={styles.actionButtonIcon} />
              </ActionButton.Item>
            </ActionButton>
          </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
    },
    actionButtonIcon: {
        fontSize: 20,
        height: 22,
        color: 'white',
    },
});
