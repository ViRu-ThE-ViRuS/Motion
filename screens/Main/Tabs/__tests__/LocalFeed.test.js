import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import LocalFeed from './../LocalFeed';

describe('Local Feed tab tests', () => {
    let tree;
    let wrapper;
    let mockNavigation;

    beforeEach(() => {
        mockNavigation = {
            navigate: jest.fn((route) => { }),
        };

        wrapper = shallow(<LocalFeed navigation={mockNavigation} />);

        tree = renderer.create(
          <LocalFeed navigation={mockNavigation} />
        );
    });

    it('renders correctly', () => {
        expect(tree).toMatchSnapshot();
    });

    it('removes backhandler correctly', () => {
        expect(tree.getInstance().componentWillUnmount()).toBe(undefined);
    });

    it('navigates correctly', () => {
        expect(tree.getInstance().createMotionScreen()).toBe(undefined);
    });
});
