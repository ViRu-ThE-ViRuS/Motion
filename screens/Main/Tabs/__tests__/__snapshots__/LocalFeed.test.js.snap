// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Local Feed tab tests renders correctly 1`] = `
<View
  style={Object {}}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Local
  </Text>
</View>
`;
